docs-source/src/architecture.rst

```diff
@@ -39,12 +65,15 @@ Grace time
 **********
 A job can take some time to check out the license from the license server after it is submitted to Slurm.
 Thus, each license has a ``grace time`` period that is used to indicate how long a booking will be retained.
-After the ``grace time`` expires, the booking is deleted. This means that the license was checked out from the
-license server and doesn't need a booking anymore.
+After the ``grace time`` expires, the booking is deleted. If the ``grace time`` expires the job is assumed to have failed
+and deemed not to consume the requested licenses, hence the booking is deleted to allow for other jobs start. Note that
+the job may still consume compute resources until the run-time expires and will from the cluster counter perspective consume
+licenses.
```

Comment:
I'm trying to think of a way to reword this, because it seems a bit unclear. I'm wondering if others have some thoughts here.

Reply:
I think this is an important aspect to cover and please propose how it can expressed better.